monix-catnap/src/main/scala/com/avast/sst/monix/catnap/CircuitBreakerModule.scala

```diff
@@ -0,0 +1,14 @@
+package com.avast.sst.monix.catnap
+
+import cats.effect.{Clock, Sync}
+import monix.catnap.CircuitBreaker
+
+object CircuitBreakerModule {
+
+  /** Makes [[monix.catnap.CircuitBreaker]] initialized with the given config and optionally [[cats.effect.Clock]]. */
+  def make[F[_]: Sync](config: CircuitBreakerConfig, clock: Option[Clock[F]] = None): F[CircuitBreaker[F]] = {
+    implicit val finalClock: Clock[F] = clock.getOrElse(Clock.create[F])
```

Comment:
I don't know if it's a good practice to create a `Clock` instance here. Don't you think we should just "force" the user to provide it always?

Reply:
Why? `Clock` gives you two methods, `monotonic` and `realTime`, so I actually think we could even simplify it and not let users pass in `Clock` but just for testing  I allow it.

Reply:
inversion of control? things with functionality, even `Clock` shouldn't be created inside functions which need them, but should be passed from above.

sooner or later you will regret if you don't do it that way :) the only thing I'm not sure of is how to pass it, there are more options
```scala
def make[F[_]: Sync: Clock](config: CircuitBreakerConfig])
```
```scala
def make[F[_]: Sync](config: CircuitBreakerConfig)(implicit clock: Clock[F])
```
```scala
def make[F[_]: Sync](config: CircuitBreakerConfig, clock: Clock[F])
```

Reply:
`def make[F[_]: Sync: Clock](config: CircuitBreakerConfig])`

Reply:
But that's what I am saying.... the only place you would need that is testing the circuit breaking which I doubt anyone will do. So my point is, I allow you to pass in your clock if you need to but most people will not care so I create a new one (which is identical to any you could create yourself because this type of clock is not timezone or anything like that specific). So we kind of get best of both worlds by using `Option`. I wouldn't pass it implicitly, so the only thing that could be changed (I don't agree because it will be much worse for the users) is to make it non-optional.

Reply:
def don't think it's a good idea to create it in here

Reply:
Could you maybe be more specific? Have you read my extensive explanations? :)

Reply:
well i assume you need F to be :Clock for the circuit breaker ... why inventing this new mechanism for providing the clock when it's already solved - just expose it to the user

Reply:
One of the design decisions of this toolkit is to use constructors for dependencies. I think `Clock` is a regular dependency, I don't see it as a lawful typeclass that should be passed implicitly using context bound. Do we agree on that or not? My view is that if I do this with `Clock` I could as well do it with `Client` (http4s client) - yes, it's possible but I don't like that approach, constructors are much more comprehensible to newcomers.

Reply:
i generally dislike the pattern of `feel free to give me something and if you don't, i will magic up something for you` ... this might be so simple that it's not an issue but still it rings alarm bells for me

Reply:
I agree but this is exactly the exception I would be willing to make. `Clock` is basically just effectful wrapper over `System.currentTimeMillis` and `System.nanoTime`, so there is nothing to configure or whatever. By making it non-optional everyone will have to create an instance manually and pass it. It just seems unnecessary in this case but if you both think it should be that way I will do it that way.

Reply:
my 2 halere: I prefer normal parameter, probably something like this
```scala
def make[F[_]: Sync](clock: Clock[F])(config: CircuitBreakerConfig)
```
but I'm OK with
```scala
def make[F[_]: Sync: Clock](config: CircuitBreakerConfig])
```

Reply:
Sorry but why making it "confusing"? Why should `Clock` be in any way different than any other dependecy? If anything this will be the end result:
```scala
def make[F[_]: Sync](config: CircuitBreakerConfig, clock: Clock[F])
```

Reply:
looking at the use of clock `CircuitBreaker.of` expects this as an implicit parameter. In that case, I would honor that style and make the signature like this:
```scala
def make[F[_]: Sync: Clock](config: CircuitBreakerConfig])
```

Reply:
Sorry but I don't want to do that. This library has its own style described in the "design decisions" documentation so I will honor that.